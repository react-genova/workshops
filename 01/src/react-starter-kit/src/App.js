import React, { Component } from 'react';

import Slider, { Slide } from './views/Slider';
import { SlideWelcome, SlideConclusions, SlideLOVS, SlideFarewell } from './views/Slides';

class App extends Component {
  render() {
    return (
        <Slider>
            <Slide title="Welcome" description="to react">
                <SlideWelcome />
            </Slide>
            <Slide title="All you need is L.O.V.S." description="List Of Valuable Suggestions">
                <SlideLOVS />
            </Slide>
            <Slide title="Conclusions" description="Yeah, interesting, so in short?">
                <SlideConclusions />
            </Slide>
            <Slide title="Farewell" description="although we're sure you will come back">
                <SlideFarewell />
            </Slide>
        </Slider>
    );
  }
}

export default App;
