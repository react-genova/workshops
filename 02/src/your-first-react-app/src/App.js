import React, { Component } from 'react';

import Slider, { Slide } from './views/Slider';
import {
    SlideLogo, SlideWhatIsReact, SlideFlexible, SlideEfficient, SlideFarewell, SlideLetsCode
} from './views/Slides';

class App extends Component {
    render() {
        return (
            <Slider>
                <SlideLogo />
                <Slide title="O React, React!" description="wherefore art thou React?">
                    <SlideWhatIsReact />
                </Slide>
                <Slide title="Flexible and Declarative" description="the odd couple">
                    <SlideFlexible />
                </Slide>
                <Slide title="Efficient" description="performance anxiety? here's the cure">
                    <SlideEfficient />
                </Slide>
                <Slide title="Let's see some code" description="where's my first react app?">
                    <SlideLetsCode />
                </Slide>
                <Slide title="Farewell" description="any questions?">
                    <SlideFarewell />
                </Slide>
            </Slider>
        );
    }
}

export default App;
