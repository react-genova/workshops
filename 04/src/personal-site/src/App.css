.app {
    overflow: hidden;
    -ms-overflow-style: -ms-autohiding-scrollbar;
    user-select: none;    
}