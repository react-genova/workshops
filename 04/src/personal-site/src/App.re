[%bs.raw {|require("./App.css")|}];

let component = ReasonReact.statelessComponent("App");

let make = (_children) => {
  ...component,
  render: _self =>
      <div className="app">
        <PageWelcome />
        <PageTechSkills />
        <PageSoftSkills />
      </div>
};
