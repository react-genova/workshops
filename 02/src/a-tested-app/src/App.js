import React, { Component } from 'react';
import logo from './logo.svg';
import MovieContainer from './views/movies/containers/MovieContainer'
import {Container, Header,Divider,Segment} from 'semantic-ui-react'
import 'semantic-ui-css/semantic.min.css'
import './App.css';

class App extends Component {
  render() {
    return (
      <div>
        <Segment
              inverted
              textAlign='center'
              style={{ minHeight: '5em', padding: '1em 0em' }}
            >
            <Header>A tested app</Header>
        </Segment>
        <Container>
          <MovieContainer />
        </Container>
      </div>
    )
  }
}

export default App;