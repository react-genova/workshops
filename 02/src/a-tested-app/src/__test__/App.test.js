import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import App from '../App';

jest.mock('../views/movies/containers/MovieContainer')
const MovieContainer = require('../views/movies/containers/MovieContainer').default;
MovieContainer.mockImplementation(() => null);

it('App snapshot', () => {
  const component = shallow(
    <App />
  );
  expect(component).toMatchSnapshot();
});

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
});

