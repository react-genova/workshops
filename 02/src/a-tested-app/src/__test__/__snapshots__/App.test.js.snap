// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot 1`] = `
<div>
  <Segment
    inverted={true}
    style={
      Object {
        "minHeight": "5em",
        "padding": "1em 0em",
      }
    }
    textAlign="center"
  >
    <Header>
      A tested app
    </Header>
  </Segment>
  <Container>
    <Refetch.connect(MovieContainer) />
  </Container>
</div>
`;
